Observer/CancelOrderItemObserver.php

```diff
@@ -3,178 +3,61 @@
-
-    /**
-     * @var DefaultSourceProviderInterface
-     */
-    private $defaultSourceProvider;
-
-    /**
-     * @var ItemToDeductFactory
-     */
-    private $itemToDeductFactory;
-
-    /**
-     * @var SourceDeductionRequestFactory
+     * @var ExecuteSourceDeductionForItems
      */
-    private $sourceDeductionRequestFactory;
+    private $executeSourceDeductionForItems;
 
-    /**
-     * @var SourceDeductionService
-     */
-    private $sourceDeductionService;
 
     /**
      * CancelOrderItemObserver constructor.
-     * @param Processor $priceIndexer
-     * @param SalesEventInterfaceFactory $salesEventFactory
-     * @param PlaceReservationsForSalesEventInterface $placeReservationsForSalesEvent
-     * @param SalesChannelInterfaceFactory $salesChannelFactory
-     * @param WebsiteRepositoryInterface $websiteRepository
      * @param GetItemsToCancelFromOrderItem $getItemsToCancelFromOrderItem
-     * @param IsSingleSourceModeInterface $isSingleSourceMode
-     * @param DefaultSourceProviderInterface $defaultSourceProvider
-     * @param ItemToDeductFactory $itemToDeductFactory
-     * @param SourceDeductionService $sourceDeductionService
+     * @param ExecuteSourceDeductionForItems $executeSourceDeductionForItems
      */
     public function __construct(
-        Processor $priceIndexer,
-        SalesEventInterfaceFactory $salesEventFactory,
-        PlaceReservationsForSalesEventInterface $placeReservationsForSalesEvent,
-        SalesChannelInterfaceFactory $salesChannelFactory,
-        WebsiteRepositoryInterface $websiteRepository,
         GetItemsToCancelFromOrderItem $getItemsToCancelFromOrderItem,
-        IsSingleSourceModeInterface $isSingleSourceMode,
-        DefaultSourceProviderInterface $defaultSourceProvider,
-        ItemToDeductFactory $itemToDeductFactory,
-        SourceDeductionRequestFactory $sourceDeductionRequestFactory,
-        SourceDeductionService $sourceDeductionService
+        ExecuteSourceDeductionForItems $executeSourceDeductionForItems
     ) {
-        $this->priceIndexer = $priceIndexer;
-        $this->salesEventFactory = $salesEventFactory;
-        $this->placeReservationsForSalesEvent = $placeReservationsForSalesEvent;
-        $this->salesChannelFactory = $salesChannelFactory;
-        $this->websiteRepository = $websiteRepository;
         $this->getItemsToCancelFromOrderItem = $getItemsToCancelFromOrderItem;
-        $this->isSingleSourceMode = $isSingleSourceMode;
-        $this->defaultSourceProvider = $defaultSourceProvider;
-        $this->itemToDeductFactory = $itemToDeductFactory;
-        $this->sourceDeductionRequestFactory = $sourceDeductionRequestFactory;
-        $this->sourceDeductionService = $sourceDeductionService;
+        $this->executeSourceDeductionForItems = $executeSourceDeductionForItems;
     }
 
     /**
      * @param EventObserver $observer
      * @return void
      */
-    public function execute(EventObserver $observer): void
```

Comment:
How come we're removing the return type here?

Reply:
For consistency with other observers..
I added the return type for others now 👍